workspace_server/src/main/java/gov/nasa/jpl/aerie/workspace/server/WorkspaceBindings.java

```diff
@@ -0,0 +1,454 @@
+package gov.nasa.jpl.aerie.workspace.server;
+
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import gov.nasa.jpl.aerie.workspace.server.postgres.NoSuchWorkspaceException;
+import io.javalin.Javalin;
+import io.javalin.apibuilder.ApiBuilder;
+import io.javalin.http.ContentType;
+import io.javalin.http.Context;
+import io.javalin.http.UnauthorizedResponse;
+import io.javalin.plugin.Plugin;
+import io.javalin.validation.ValidationException;
+
+import javax.json.Json;
+import javax.json.JsonException;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.file.Path;
+import java.sql.SQLException;
+import java.util.Objects;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.javalin.apibuilder.ApiBuilder.path;
+
+public class WorkspaceBindings implements Plugin {
+  private static final Logger logger = LoggerFactory.getLogger(WorkspaceBindings.class);
+  private final JWTService jwtService;
+  private final WorkspaceService workspaceService;
+
+  public WorkspaceBindings(final JWTService jwtService, final WorkspaceService workspaceService) {
+    this.jwtService = jwtService;
+    this.workspaceService = workspaceService;
+  }
+
+  private record PathInformation(int workspaceId, Path filePath) {
+    static PathInformation of(Context context) {
+      final var workspaceId = Integer.parseInt(context.pathParam("workspaceId"));
+      final var filePath = Path.of(context.pathParam("filePath"));
+
+      return new PathInformation(workspaceId, filePath);
+    }
+
+    String fileName() {
+      return filePath.getFileName().toString();
+    }
+  }
+
+  @Override
+  public void apply(final Javalin javalin) {
+    javalin.routes(() -> {
+      //before("/ws/*", authorize); <- dont want to force auth on the health check.
```

Comment:
This is commented out, but without it, we don't auth users (except on the `createWorkspace` route). We think this should be:
```
before("/ws/*", this::authorize);
```
does that seem right @Mythicaeda ?

Reply:
Oh, it's commented out because there was no implemented `authorize` method when I drafted it. I'll uncomment it, just be aware that it's currently only checking that an unexpired, valid JWT was provided ("valid" meaning the user can access the role they're using and that the signature is correct), as opposed to a detailed permissions check.